lib/src/collection/bimap.dart

```diff
@@ -119,14 +117,16 @@ class HashBiMap<K, V> implements BiMap<K, V> {
 
   @override
   V putIfAbsent(K key, V ifAbsent()) {
-    var value = _map[key];
-    if (value != null) return value;
-    if (!_map.containsKey(key)) return _add(key, ifAbsent(), false);
-    return null;
+    if (_map.containsKey(key)) {
+      return _map[key]!;
+    }
```

Comment:
Perhaps it doesn't matter, but the previous version was faster because in the case when the key is present we did one map lookup. The new version performs two lookups (`containsKey` then `[key]`).

Is the change because the nullness of the value is not equivalent to missing key any more?

Reply:
The previous version was correct, but the return null at the end was dead code except with Bimap<T?> which doesn't make sense.

Reply:
FWIW, the original implementation works with a little tweaking. Maybe adding a comment fixes the readability concern.
```dart
@override                                                                     
V putIfAbsent(K key, V ifAbsent()) {                                          
  V? value = _map[key];                                                       
  if (value != null) return value;                                            
  if (!_map.containsKey(key)) return _add(key, ifAbsent(), false);            
  // Value is null, which is disallowed.                                                                     
  return value!;                                                              
}                                                                             
```

Reply:
Re-pushed with that implementation.

Reply:
And re-pushed with an even better implementation. Nullable types are now disallowed for K and V so we know if V is null that K is not a key.

Reply:
Added a second commit as a "what do you think?"

There's really no reason we shouldn't allow nullable keys or values, subject to the same constraint that we've always had, that each value in the map be unique (since it's used as a key in the inverse), we'd simply be allowing one more legal value -- a single null.